<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="0" failed="2" ignored="0" total="4" passed="2">
  <reporter-output>
    <line>
      <![CDATA[[INFO]: Title Found: Get Latest News, India News, Breaking News, Today's News - NDTV.com]]>
    </line>
    <line>
      <![CDATA[[ASSERT PASSED]: Page title found on Home page is correct.]]>
    </line>
    <line>
      <![CDATA[[INFO]: Title Found: NDTV Weather - Weather in your Indian City]]>
    </line>
    <line>
      <![CDATA[[ASSERT PASSED]: Page title found on Weather page is correct.]]>
    </line>
    <line>
      <![CDATA[[ASSERT FAILED]: Element cityCheckbox not found on the WeatherPage !!!]]>
    </line>
    <line>
      <![CDATA[[ASSERT FAILED]: Element mapCity not found on the WeatherPage !!!]]>
    </line>
  </reporter-output>
  <suite name="Default suite" duration-ms="56177" started-at="2021-07-02T18:29:51Z" finished-at="2021-07-02T18:30:47Z">
    <groups>
    </groups>
    <test name="Default test" duration-ms="56177" started-at="2021-07-02T18:29:51Z" finished-at="2021-07-02T18:30:47Z">
      <class name="automation.tests.Weather_Verification_Test">
        <test-method status="PASS" signature="start_test_session(java.lang.String)[pri:0, instance:automation.tests.Weather_Verification_Test@464bee09]" name="start_test_session" is-config="true" duration-ms="4727" started-at="2021-07-02T18:29:51Z" finished-at="2021-07-02T18:29:56Z">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- start_test_session -->
        <test-method status="PASS" signature="Test01_Launch_Application_And_Verify_User_Navigation_To_HomePage()[pri:0, instance:automation.tests.Weather_Verification_Test@464bee09]" name="Test01_Launch_Application_And_Verify_User_Navigation_To_HomePage" duration-ms="47632" started-at="2021-07-02T18:29:56Z" finished-at="2021-07-02T18:30:44Z">
          <reporter-output>
            <line>
              <![CDATA[[INFO]: Title Found: Get Latest News, India News, Breaking News, Today's News - NDTV.com]]>
            </line>
            <line>
              <![CDATA[[ASSERT PASSED]: Page title found on Home page is correct.]]>
            </line>
          </reporter-output>
        </test-method> <!-- Test01_Launch_Application_And_Verify_User_Navigation_To_HomePage -->
        <test-method status="PASS" signature="Test02_Click_On_Weather_Tab_And_Verify_User_Navigates_To_Weather_Page()[pri:0, instance:automation.tests.Weather_Verification_Test@464bee09]" name="Test02_Click_On_Weather_Tab_And_Verify_User_Navigates_To_Weather_Page" duration-ms="1677" started-at="2021-07-02T18:30:44Z" finished-at="2021-07-02T18:30:45Z">
          <reporter-output>
            <line>
              <![CDATA[[INFO]: Title Found: NDTV Weather - Weather in your Indian City]]>
            </line>
            <line>
              <![CDATA[[ASSERT PASSED]: Page title found on Weather page is correct.]]>
            </line>
          </reporter-output>
        </test-method> <!-- Test02_Click_On_Weather_Tab_And_Verify_User_Navigates_To_Weather_Page -->
        <test-method status="FAIL" signature="Test03_enter_City_And_Verify_City_Is_Displayed_In_Map()[pri:0, instance:automation.tests.Weather_Verification_Test@464bee09]" name="Test03_enter_City_And_Verify_City_Is_Displayed_In_Map" duration-ms="579" started-at="2021-07-02T18:30:45Z" finished-at="2021-07-02T18:30:46Z">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[[ASSERT FAILED]: Element cityCheckbox not found on the WeatherPage !!!]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: [ASSERT FAILED]: Element cityCheckbox not found on the WeatherPage !!!
at org.testng.Assert.fail(Assert.java:96)
at automation.base.GetPage.elemconstructed_dynamically(GetPage.java:58)
at automation.keywords.WeatherPage.selectCheckboxForCityIfNotSelected(WeatherPage.java:39)
at automation.keywords.WeatherPage.enterCityInPinYourCityTextFieldAndSelect(WeatherPage.java:35)
at automation.tests.Weather_Verification_Test.Test03_enter_City_And_Verify_City_Is_Displayed_In_Map(Weather_Verification_Test.java:38)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
at org.testng.TestRunner.privateRun(TestRunner.java:648)
at org.testng.TestRunner.run(TestRunner.java:505)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
at org.testng.SuiteRunner.run(SuiteRunner.java:364)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
at org.testng.TestNG.runSuites(TestNG.java:1049)
at org.testng.TestNG.run(TestNG.java:1017)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:114)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
            <line>
              <![CDATA[[ASSERT FAILED]: Element cityCheckbox not found on the WeatherPage !!!]]>
            </line>
          </reporter-output>
        </test-method> <!-- Test03_enter_City_And_Verify_City_Is_Displayed_In_Map -->
        <test-method status="FAIL" signature="Test04_Verify_City_Weather_Information_Displayed_In_Map()[pri:0, instance:automation.tests.Weather_Verification_Test@464bee09]" name="Test04_Verify_City_Weather_Information_Displayed_In_Map" duration-ms="52" started-at="2021-07-02T18:30:46Z" finished-at="2021-07-02T18:30:46Z">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[[ASSERT FAILED]: Element mapCity not found on the WeatherPage !!!]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: [ASSERT FAILED]: Element mapCity not found on the WeatherPage !!!
at org.testng.Assert.fail(Assert.java:96)
at automation.base.GetPage.elemconstructed_dynamically(GetPage.java:58)
at automation.keywords.WeatherPage.storeCityWeatherInformation(WeatherPage.java:51)
at automation.tests.Weather_Verification_Test.Test04_Verify_City_Weather_Information_Displayed_In_Map(Weather_Verification_Test.java:44)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
at org.testng.TestRunner.privateRun(TestRunner.java:648)
at org.testng.TestRunner.run(TestRunner.java:505)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
at org.testng.SuiteRunner.run(SuiteRunner.java:364)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
at org.testng.TestNG.runSuites(TestNG.java:1049)
at org.testng.TestNG.run(TestNG.java:1017)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:114)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
            <line>
              <![CDATA[[ASSERT FAILED]: Element mapCity not found on the WeatherPage !!!]]>
            </line>
          </reporter-output>
        </test-method> <!-- Test04_Verify_City_Weather_Information_Displayed_In_Map -->
        <test-method status="PASS" signature="stop_test_session()[pri:0, instance:automation.tests.Weather_Verification_Test@464bee09]" name="stop_test_session" is-config="true" duration-ms="1470" started-at="2021-07-02T18:30:46Z" finished-at="2021-07-02T18:30:47Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- stop_test_session -->
      </class> <!-- automation.tests.Weather_Verification_Test -->
    </test> <!-- Default test -->
  </suite> <!-- Default suite -->
</testng-results>
